### PR TITLE
grpcgen: skip clusters with no service 

### PR DESCRIPTION
**Pull request by stevenctl**
https://github.com/istio/istio/issues/44701

### CODE REVIEW

File: pilot/pkg/autoregistration/controller.go [hunk 1]
```diff
@@ -229,15 +229,26 @@ func (c *Controller) RegisterWorkload(proxy *model.Proxy, conTime time.Time) err
 }
 
 func (c *Controller) registerWorkload(entryName string, proxy *model.Proxy, conTime time.Time) error {
+	groupCfg := c.store.Get(gvk.WorkloadGroup, proxy.Metadata.AutoRegisterGroup, proxy.Metadata.Namespace)
```
Comment (howardjohn) [inline pilot/pkg/autoregistration/controller.go:232]: is this meant for this PR?

**Review by howardjohn — COMMENTED** [review]
lgtm except WE stuff?

File: pilot/pkg/networking/grpcgen/cds.go [hunk 2]
```diff
@@ -103,13 +103,13 @@ func newClusterBuilder(node *model.Proxy, push *model.PushContext, defaultCluste
...
-		var ok bool
-		port, ok = svc.Ports.GetByPort(portNum)
-		if !ok {
-			log.Warnf("cds gen for %s: did not find port %d in service for cluster %s", node.ID, portNum, defaultClusterName)
-		}
+		return nil, fmt.Errorf("cds gen for %s: did not find service for cluster %s", node.ID, defaultClusterName)
+	}
+
+	var ok bool
+	port, ok = svc.Ports.GetByPort(portNum)
```
Comment (howardjohn) [inline pilot/pkg/networking/grpcgen/cds.go:110]: nit:
```suggestion
	port, ok := svc.Ports.GetByPort(portNum)
```

with refactoring can be simplified

File: pilot/pkg/autoregistration/controller.go (hunk 1 above)
Comment (stevenctl) [reply pilot/pkg/autoregistration/controller.go]: nope

**Pull request merged**